CODE_STYLE.md

```diff
@@ -0,0 +1,3293 @@
+  end
+
+  # bad (double indent)
+  def send_mail(source)
+    Mailer.deliver(
+        to: 'bob@example.com',
+        from: 'us@example.com',
+        subject: 'Important message',
+        body: source.text)
+  end
+
+  # good
+  def send_mail(source)
+    Mailer.deliver(to: 'bob@example.com',
+                   from: 'us@example.com',
+                   subject: 'Important message',
+                   body: source.text)
+  end
+
+  # good (normal indent)
+  def send_mail(source)
+    Mailer.deliver(
+      to: 'bob@example.com',
+      from: 'us@example.com',
+      subject: 'Important message',
+      body: source.text
+    )
+  end
+  ```
+
+* Align the elements of array literals spanning multiple lines.
+
+  ```Ruby
+  # bad - single indent
+  menu_item = ['Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam',
+    'Baked beans', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam']
+
+  # good
+  menu_item = [
+    'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam',
+    'Baked beans', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam'
+  ]
+
+  # good
+  menu_item =
+    ['Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam',
+     'Baked beans', 'Spam', 'Spam', 'Spam', 'Spam', 'Spam']
+  ```
+
+* Add underscores to large numeric literals to improve their readability.
+
+  ```Ruby
+  # bad - how many 0s are there?
+  num = 1000000
+
+  # good - much easier to parse for the human brain
+  num = 1_000_000
+  ```
+
+ * Use RDoc and its conventions for API documentation.  Don't put an empty line between the comment block and the `def`.
+
+  * Limit lines to 80 characters.
+  * Avoid trailing whitespace.
+  * End each file with a newline.
+  * Don't use block comments. They cannot be preceded by whitespace and are not as easy to spot as regular comments.
+
+  ```Ruby
+  # bad
+  =begin
+  comment line
+  another comment line
+  =end
+
+  # good
+  # comment line
+  # another comment line
+  ```
+
+## Syntax
+
+* <a name="double-colons"></a>
+    Use `::` only to reference constants(this includes classes and
+    modules) and constructors (like `Array()` or `Nokogiri::HTML()`).
+    Do not use `::` for regular method invocation.
+<sup>[[link](#double-colons)]</sup>
+
+  ```Ruby
+  # bad
+  SomeClass::some_method
+  some_object::some_method
+
+  # good
+  SomeClass.some_method
+  some_object.some_method
+  SomeModule::SomeClass::SOME_CONST
+  SomeModule::SomeClass()
+  ```
+
+* <a name="method-parens"></a>
+    Use `def` with parentheses when there are parameters. Omit the
+    parentheses when the method doesn't accept any parameters.
+<sup>[[link](#method-parens)]</sup>
+
+   ```Ruby
+   # bad
+   def some_method()
+     # body omitted
+   end
+
+   # good
+   def some_method
+     # body omitted
+   end
+
+   # bad
+   def some_method_with_parameters param1, param2
+     # body omitted
+   end
+
+   # good
+   def some_method_with_parameters(param1, param2)
+     # body omitted
+   end
+   ```
```

Comment:
I just want to stress that I am strongly in favour of strict compliance to 
```ruby
def some_method
```
and 
```ruby
def some_method(param1, param2)
```
same for invokation
```ruby
some_method
```
and
```ruby
some_method(param1, param2)
```
my only 2 exceptions are `raise` and `return`.